### PR TITLE
Add API test case assign user system admin role

### PR DESCRIPTION
**Pull request by danfengliu**
Add API test case assign user system admin role
Signed-off-by: danfengliu <danfengl@vmware.com>

### CODE REVIEW

File: tests/apitests/python/library/configurations.py [hunk 1]
```diff
@@ -2,25 +2,77 @@
...
 import swagger_client
+from swagger_client.rest import ApiException
+
+def set_configurations(client, expect_status_code = 200, expect_response_body = None, **config):
+    conf = swagger_client.Configurations()
+    if "project_creation_restriction" in config:
+        conf.project_creation_restriction = config.get("project_creation_restriction")
+    if "token_expiration" in config:
+        conf.token_expiration = config.get("token_expiration")
+    HasError = False
```
Comment (clouderati) [inline tests/apitests/python/library/configurations.py]: Would `has_error` make more sense given the other variable names in this particular file (e.g., `token_expiration`)?

File: tests/apitests/python/library/configurations.py [hunk 2]
```diff
@@ -2,25 +2,77 @@
...
+    try:
+        _, status_code, _ = client.configurations_put_with_http_info(conf)
+    except ApiException as e:
+        HasError = True
```
Comment (clouderati) [inline tests/apitests/python/library/configurations.py]: Ditto above

File: tests/apitests/python/library/configurations.py [hunk 3]
```diff
@@ -2,25 +2,77 @@
...
+        if e.status == expect_status_code:
+            if expect_response_body is not None and e.body.strip() != expect_response_body.strip():
+                raise Exception(r"Set configuration response body is not as expected {} actual status is {}.".format(expect_response_body.strip(), e.body.strip()))
```
Comment (clouderati) [inline tests/apitests/python/library/configurations.py]: nit: rephrase as follows:

`Set configuration response body is not as we expected. Expected {}, while actual status is {}`.

File: tests/apitests/python/library/configurations.py [hunk 4]
```diff
@@ -2,25 +2,77 @@
...
+            else:
+                return e.reason, e.body
+        else:
+            raise Exception(r"Set configuration result is not as expected {} actual status is {}.".format(expect_status_code, e.status))
```
Comment (clouderati) [inline tests/apitests/python/library/configurations.py]: Ditto above

File: tests/apitests/python/library/configurations.py [hunk 5]
```diff
@@ -2,25 +2,77 @@
...
         client = self._get_client(**kwargs)
-        data, status_code, _ = client.configurations_get_with_http_info()
+        HasError = False
+        try:
+            data, status_code, _ = client.configurations_get_with_http_info()
+        except ApiException as e:
+            HasError = True
+            if e.status == expect_status_code:
+                if expect_response_body is not None and e.body.strip() != expect_response_body.strip():
+                    raise Exception(r"Get configuration response body is not as expected {} actual status is {}.".format(expect_response_body.strip(), e.body.strip()))
```
Comment (clouderati) [inline tests/apitests/python/library/configurations.py]: Ditto above

File: tests/apitests/python/library/configurations.py [hunk 6]
```diff
@@ -2,25 +2,77 @@
...
+                else:
+                    return e.reason, e.body
+            else:
+                raise Exception(r"Get configuration result is not as expected {} actual status is {}.".format(expect_status_code, e.status))
```
Comment (clouderati) [inline tests/apitests/python/library/configurations.py]: Ditto above

File: tests/apitests/python/library/user.py [hunk 7]
```diff
@@ -91,4 +91,9 @@ def update_uesr_role_as_sysadmin(self, user_id, IsAdmin, **kwargs):
         print "has_admin_role:", has_admin_role
         _, status_code, _ = client.users_user_id_sysadmin_put_with_http_info(user_id, has_admin_role)
         base._assert_status_code(200, status_code)
-        return user_id
+
+    def update_uesr_role_as_sysadmin_success(self, user_id, IsAdmin, **kwargs):
+        self.update_uesr_role_as_sysadmin(user_id, IsAdmin, **kwargs)
+        user = self.get_user(user_id, **kwargs)
+        if user.has_admin_role != IsAdmin:
+            raise Exception(r"Update user role as sysadmin failed!")
```
Comment (clouderati) [inline tests/apitests/python/library/user.py]: nit: remove `!` in exception

File: tests/apitests/python/test_assign_sys_admin.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        self.conf= conf
+
+        user = User()
+        self.user= user
+
+    @classmethod
+    def tearDown(self):
+        print "Case completed"
+
+    @unittest.skipIf(TEARDOWN == False, "Test data should be remain in the harbor.")
```
Comment (clouderati) [inline tests/apitests/python/test_assign_sys_admin.py]: Rephrase message

File: tests/apitests/python/test_assign_sys_admin.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    def test_ClearData(self):
+        #1. Delete user(UA);
+        self.user.delete_user(TestProjects.user_assign_sys_admin_id, **ADMIN_CLIENT)
+
+    def testAssignSysAdmin(self):
+        """
+        Test case:
+            Assign Sys Admin
```
Comment (clouderati) [inline tests/apitests/python/test_assign_sys_admin.py]: s/Sys Admin/sysadmin

File: tests/apitests/python/test_assign_sys_admin.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        Test step & Expectation:
```
Comment (clouderati) [inline tests/apitests/python/test_assign_sys_admin.py]: Rephrase: `Test step and expected result`

Comment (danfengliu) [reply tests/apitests/python/test_assign_sys_admin.py]: Thanks, corrected.

File: tests/apitests/python/library/configurations.py (hunk 1 above)
Comment (danfengliu) [reply tests/apitests/python/library/configurations.py]: Thanks, I renamed has_error to caught_api_exception, is it good?

File: tests/apitests/python/library/configurations.py (hunk 2 above)
Comment (danfengliu) [reply tests/apitests/python/library/configurations.py]: Ditto above too.

File: tests/apitests/python/library/configurations.py (hunk 3 above)
Comment (danfengliu) [reply tests/apitests/python/library/configurations.py]: Thanks, corrected.

File: tests/apitests/python/library/configurations.py (hunk 4 above)
Comment (danfengliu) [reply tests/apitests/python/library/configurations.py]: Thanks, corrected.

File: tests/apitests/python/library/configurations.py (hunk 5 above)
Comment (danfengliu) [reply tests/apitests/python/library/configurations.py]: Thanks, corrected.

File: tests/apitests/python/library/configurations.py (hunk 6 above)
Comment (danfengliu) [reply tests/apitests/python/library/configurations.py]: Thanks, corrected.

File: tests/apitests/python/library/user.py (hunk 7 above)
Comment (danfengliu) [reply tests/apitests/python/library/user.py]: Thanks, corrected.

File: tests/apitests/python/test_assign_sys_admin.py (hunk 8 above)
Comment (danfengliu) [reply tests/apitests/python/test_assign_sys_admin.py]: Thanks, does "Test data should be remain in the harbor." sounds good?

File: tests/apitests/python/test_assign_sys_admin.py (hunk 9 above)
Comment (danfengliu) [reply tests/apitests/python/test_assign_sys_admin.py]: Hi clouderati, could you tell me more about this issue, I'm not quite clear on it.

Comment (danfengliu) [reply tests/apitests/python/test_assign_sys_admin.py]: @clouderati  Thank you, I still have one issue, and need your input.

**Comment by wy65701436** [comment]
@clouderati could you please help to review the changes again? The new test case development depends on this PR.

**Comment by danfengliu** [comment]
This request is not in-use.

**Pull request closed without merging**